lib/iterator.js

```diff
@@ -249,6 +253,28 @@ class TakeIterator extends Iterator {
   }
 }
 
+class TakeWhileIterator extends Iterator {
+  constructor(base, predicate, thisArg) {
+    super(base);
+    this.predicate = predicate;
+    this.thisArg = thisArg;
+    this.done = false;
+  }
+
+  next() {
+    if (this.done) return { done: true, value: undefined };
+    for (const value of this.base) {
```

Comment:
I don't think you need a loop here since you are always making only one iteration.

Reply:
I don't think I understand, I iterate over the base iterator, therefore, this loop will iterate over all of base-iterator values untill `predicate` return false.

Reply:
You only have an `if` statement inside this loop and in each of the `if` statement's branches you exit the loop, meaning that you always make only one iteration.

Reply:
Oh, good catch. I just copy-pasted Filter impl here :upside_down_face:.